src/main/java/at/ac/tuwien/kr/alpha/common/predicates/TotalOrder.java

```diff
@@ -0,0 +1,85 @@
+package at.ac.tuwien.kr.alpha.common.predicates;
+
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
+
+import java.util.List;
+
+public class TotalOrder implements Evaluable, Predicate {
+	private final String predicateName;
+
+	public TotalOrder(String predicateName) {
+		this(predicateName, false);
+	}
+
+	public TotalOrder(String predicateName, boolean flip) {
+		if (!flip) {
+			this.predicateName = predicateName;
+			return;
+		}
+
+		switch (predicateName) {
+			case "=":
```

Comment:
I'd prefer the BINOP enum being used here, since this enum occurs at several positions (aggregates, choice rules) and the code seems to be better typed in the end than with strings.

Reply:
OK